typescript/testsuite/testsuite_impl/basic_datastore_and_api_test/basic_datastore_and_api_test_.ts

```diff
@@ -0,0 +1,238 @@
+const DATASTORE_SERVICE_ID: ServiceID = "datastore";
+const DATASTORE_PORT: number = 1323;
+
+const API_SERVICE_IMAGE: string = "kurtosistech/example-microservices_api";
+const API_SERVICE_ID: ServiceID = "api";
+const API_SERVICE_PORT: number = 2434;
+
+const WAIT_FOR_STARTUP_DELAY_MILLISECONDS: number = 1000;
+const WAIT_FOR_STARTUP_MAX_POLLS: number = 15;
+
+const TEST_PERSON_ID: number = 23;
+const TEST_NUM_BOOKS_READ: number = 3;
+
+const CONFIG_FILE_KEY: string  = "config-file";
+
+
+class DatastoreConfig {
+	private readonly datastoreIp: string;
+	private readonly datastorePort: number;
+	
+	constructor(datastoreIp: string, datastorePort: number) {
+		this.datastoreIp = datastoreIp;
+		this.datastorePort = datastorePort;
+	}
+}
+
+export class BasicDatastoreAndApiTest {
+    private readonly datastoreImage: string;
+    private readonly apiImage: string;
+    
+    constructor(datastoreImage: string, apiImage: string) {
+		this.datastoreImage = datastoreImage;
+		this.apiImage = apiImage;
+    }
+	
+	public configure(builder: TestConfigurationBuilder): void {
+		builder.withSetupTimeoutSeconds(60).withRunTimeoutSeconds(60); //TODO (Ali) - allowed since typescript gives direct reference
+    }
+
+	public async setup(networkCtx: NetworkContext): Promise<Result<Network, Error>> { //TODO (Ali) - async?
+		
+		const [datastoreContainerCreationConfig, datastoreRunConfigFunc]: [ContainerCreationConfig, (ipAddr: string, generatedFileFilepaths: Map<string, string>, staticFileFilepaths: Map<StaticFileID, string>) => Result<ContainerRunConfig, Error>] = await getDatastoreServiceConfigurations(); //TODO (Ali) - maybe Result here
+
+		const addDatastoreServiceResult: Result<[ServiceContext, Map<string, PortBinding>], Error> = await networkCtx.addService(DATASTORE_SERVICE_ID, datastoreContainerCreationConfig, datastoreRunConfigFunc);
+		if (!addDatastoreServiceResult.isOk()) {
+			return err(addDatastoreServiceResult.error)
+		}
+		const [datastoreServiceContext, datastoreSvcHostPortBindings]: [ServiceContext, Map<string, PortBinding>] = addDatastoreServiceResult.value;
+
+		const datastoreClient: DatastoreClient = new DatastoreClient(datastoreServiceContext.getIPAddress(), DATASTORE_PORT);
+
+		const datastoreWaitForHealthyResult: Result<null, Error> = await datastoreClient.waitForHealthy(WAIT_FOR_STARTUP_MAX_POLLS, WAIT_FOR_STARTUP_DELAY_MILLISECONDS);
+		if (!datastoreWaitForHealthyResult.isOk()) {
+			return err(datastoreWaitForHealthyResult.error);
+		}
+
+		log.info("Added datastore service with host port bindings: %+v", datastoreSvcHostPortBindings);
+
+		const [apiServiceContainerCreationConfig, apiServiceRunConfigFunc]: [ContainerCreationConfig, (ipAddr: string, generatedFileFilepaths: Map<string, string>, staticFileFilepaths: Map<StaticFileID, string>) => Result<ContainerRunConfig, Error>] = await getApiServiceConfigurations(datastoreClient) //TODO (Ali) - add maybe Result here
```

Comment:
Same thing here - could inline the `getApiServiceConfigurations` contents and get rid of it to clean this up